purchase_order_general_discount/models/purchase_order.py

```diff
@@ -0,0 +1,78 @@
+# Copyright 2019 Tecnativa - David Vidal
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from lxml import etree
+
+from odoo import api, fields, models
+
+
+class PurchaseOrder(models.Model):
+    _inherit = "purchase.order"
+
+    general_discount = fields.Float(
+        digits="Discount",
+        string="Gen. Disc. (%)",
+    )
+
+    _sql_constraints = [
+        (
+            "general_discount_limit",
+            "CHECK (general_discount <= 100.0)",
+            "Discount must be lower than 100%.",
+        ),
+    ]
+
+    @api.onchange("partner_id")
+    def onchange_partner_id(self):
+        super().onchange_partner_id()
+        self.general_discount = (
+            self.partner_id.commercial_partner_id.purchase_general_discount
+        )
+        return super().onchange_partner_id()
```

Comment:
This is not the way to solve the pre-commit warning. You're running super() twice here.

```suggestion
        res = super().onchange_partner_id()
        self.general_discount = (
            self.partner_id.commercial_partner_id.purchase_general_discount
        )
        return res
```

Reply:
You're right, it's already corrected, thanks for your teachings.